Create monit mongo for rucio dataset monitoring

mrceyhun opened the pull request:
Created for Rucio dataset monitoring test MongoDB instance. Migrated from CMSMonitoring dev branch.

 This is a working test deployment with 1 MongoDB instance which is deployed as `StatefulSet`. If we'll continue with this custom solution in production, I'll deploy it as 1 primary, 1 replica instances as `StatefulSet` with other prod requirements like persistent volumes, static dns in k8s cluster, etc. 

- Secrets are arranged and new secret is created in the gitlab repository (MR opened). @vkuznet  I will migrate it to "sops/age" soon.
- mongo-test-client is just for test
- There will be 2 additional services in future: 1st to serve jquery datatable (golang will serve static index.html file), 2nd to run `mongoimport` cron.

fyi @dynamic-entropy

vkuznet commented on kubernetes/monitoring/deploy-secrets.sh:
```diff
@@ -105,6 +106,14 @@ elif [ "$secret" == "log-clustering-secrets" ]; then
     cmsmonit_f=`ls $sdir/cmsmonit-keytab/ | awk '{ORS=" " ; print "--from-file="D"/"$1""}' D=$sdir/cmsmonit-keytab | sed "s, $,,g"`
     files="${log_f} ${cmsmonit_f}"
+elif [ "$secret" == "monit-mongo-secrets" ]; then
+mongo_ru=`cat $sdir/monit-mongo-secrets/secrets | grep MONGO_ROOT_USERNAME | awk '{print $2}'`
```
could you please use proper indent for code inside `elif`, it is small thing but it is easy for eyes when opened in editor.

mrceyhun replied:
Thanks Valentin, i forget to rollback a temporary change. Now I made it more readable.

vkuznet commented on kubernetes/monitoring/services/mongo/monit-mongo.yaml:
```diff
@@ -0,0 +1,164 @@
+  mongo.conf: |
+    storage:
+      dbPath: /data/db
+  ensure-users.js: |
```
it is better to put actual files into repository area and package them with the image. This will make file available and easy to debug. Here you make it as part of k8s manifest, and even though it will be created someone cannot use it until stuff deployed to k8s, while if it is part of the image, I can run docker run and debug everything within container without needs to have k8s infrastructure.

mrceyhun replied:
Thanks for the suggestion. Yes, today I dealt with a similar debugging JS case. I created a Dockerfile `monit-mongo`, built the image and pushed to cern registry. Now, JS and conf file are moved from ConfigMap to Docker image.

vkuznet commented on kubernetes/monitoring/services/mongo/monit-mongo.yaml:
```diff
@@ -0,0 +1,164 @@
+    - port: 27017
+      protocol: TCP
+      targetPort: 27017
+      nodePort: 32000
```
once again, why do you need nodePort? k8s provide excellent capabilities to reach services and we rarely need to expose service ports. Instead, you can reach your service as simple as `<svc-name>.<namesapce>.svc.cluster.local:<port>`. For instance, in auth proxy server, I use it all the time to redirect HTTP request to, e.g. `dbs2go.dbs.svc.cluster.local:8217`. I suggest that you evaluate the needs for nodePort and make proper adjustments.

vkuznet commented on kubernetes/monitoring/services/mongo/monit-mongo.yaml:
```diff
@@ -0,0 +1,164 @@
+      labels:
+        app: monit-mongo
+        selector: monit-mongo
+    spec:
```
I suggest to add `livenessProbe` section which will ensure that k8s can ping service and restart it. Also, please add `resources` section which will constrain required resources for this service. You can view how we use them in any manifest files from cmsweb area, or consult with @muhammadimranfarooqi

mrceyhun commented:
Many thanks for the review Valentin. I made all the changes and PR is ready to re-review :)

vkuznet reviewed (commented):
I still recommend to disable NodePort if it is not required for operations, and switch to mongo shell tool

amaltaro commented:
Out of curiosity, is there any reason not to use the MongoDB as a service? Which was provided by the HTTP/Panos group in the last year.

mrceyhun commented:
@vkuznet I removed the NodePort and converted to headless service. Also updated the readme with the new host k8s FQDN.

@amaltaro Thanks for the suggestion. We created this instance for test and Q&A. For production, we'll decide between the 2 options: 1-) using this custom instance after some improvements, 2-) using Panos/HTTP MongoDB service.

Panos suggested the second option too and we decided to test it when he'll come back from the vacation (AFAIK, he is in vacation now). Decided in [JIRA CMSMONIT-474](https://its.cern.ch/jira/browse/CMSMONIT-474)

vkuznet commented:
@mrceyhun do you want me to merge it?

mrceyhun commented:
Thanks a lot Valentin, I merged.
